src/ray/gcs/gcs_server/ray_sync.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+#include "ray/common/asio/periodical_runner.h"
+#include "ray/gcs/gcs_server/gcs_resource_report_poller.h"
+#include "ray/gcs/gcs_server/grpc_based_resource_broadcaster.h"
+
+namespace ray {
+namespace sync {
+
+class RaySync {
```

Comment:
nit: is it better to name it ResourceSyncer?

Reply:
It's going to be a general component. I don't want to bind it to Resource only. Later scheduling info is also going to use this component.
Do you have other suggestions for this?

Reply:
In the long term, what information will be synced through this class? (I think we should at least rename sync to syncer)

Reply:
It should be anything you think is worth broadcasting or sending.